sponsorship_compassion/models/contracts.py

```diff
@@ -532,9 +546,23 @@ def contract_active(self):
                 lambda c: 'S' in c.type and not c.is_active):
             # UpsertConstituent Message
             partner = contract.correspondant_id
-            partner.upser_constituent()
-
-            # TODO : Create Sponsorship Message
+            partner.upsert_constituent()
+
+            # Create Sponsorship Message
+            message_obj = self.env['gmc.message.pool']
+            action_id = self.env.ref(
+                'sponsorship_compassion.create_sponsorship').id
+
+            messages = message_obj.search([

```

Comment:
I have a better idea for this check : as it is general for each message (we don't wan't to create a new message if one similar does already exist), we can integrate it in the create method of object `gmc_message` and don't worry of that outside. Can you make the change ? We can discuss it if you need more precisions.